Source/missiles.cpp

```diff
@@ -1067,6 +1067,10 @@ void SetMissAnim(int mi, int animtype)
 {
 	int dir = missile[mi]._mimfnum;
 
+    if (animtype == MFILE_NONE) {
+        animtype = misfilelength - 1;

```

Comment:
would things break if we simply returned here?

Reply:
I think some things could break, but I can't verify that as I didn't copy the save game off my device after I verified the fix as it is in the PR now. I can try to reproduce this in some other conditions and get the save game for reproduction if you think that's worth the effort.

As there is valid data for `MFILE_NONE` in `misfiledata` (mostly just zeroes), it probably would make sense to zero out `missile[mi]` without early return. Returning here without zeroing out `missile[mi]` could lead to weird and hard to reproduce bugs, I guess. Also, interesting how it worked in vanilla, which I assume did OOB access and just wrote garbage (or zeroes after all) into `missile[mi]`?

Reply:
Ok, had some time to look at this. I think you might be right that a return wouldn't be the right thing to do as it's fairly explicitly calld in this way when MFILE_NONE is the case. But instead of counting the array length we can simply set it to MFILE_NULL.

Reply:
Looking a bit further into it I would say that under normal conditions it wouldn't be an issue to simply return and leave it with garbage data as the missiles of this type fall into 3 categories `draw = false`, custom sprite set during spawn (rhino), or unimplemented. But probably better to have clean data rather than later having to track down a random bug :)

This has now been merged via https://github.com/diasurgical/devilutionX/commit/aba60c560d23cc14282678b03fc00272d691bc70 plus my suggestions. The enums have also been applied upstream so will come at next merge.